packages/roosterjs-content-model-plugins/lib/imageEdit/ImageEditPlugin.ts

```diff
@@ -118,14 +130,170 @@ export class ImageEditPlugin implements ImageEditor, EditorPlugin {
      * exclusively by another plugin.
      * @param event The event to handle:
      */
-    onPluginEvent(_event: PluginEvent) {}
+    onPluginEvent(event: PluginEvent) {
+        if (!this.editor) {
+            return;
+        }
+        switch (event.eventType) {
+            case 'mouseUp':
+                this.mouseUpHandler(this.editor, event);
+                break;
+            case 'keyDown':
+                this.keyDownHandler(this.editor, event);
+                break;
+        }
+    }
+
+    private mouseUpHandler(editor: IEditor, event: MouseUpEvent) {
+        const selection = editor.getDOMSelection();
+        if ((selection && selection.type == 'image') || this.isEditing) {
+            this.applyFormatWithContentModel(
+                editor,
+                selection,
+                this.isCropMode,
+                false /* shouldSelectImage */
+            );
+        }
+    }
+
+    //Sometimes the cursor can be inside the editing image and inside shadow dom, then the cursor need to moved out of shadow dom
+    private selectBeforeEditingImage(editor: IEditor, element: HTMLElement) {
+        let parent = element.parentNode;
+        if (parent && isNodeOfType(parent, 'ELEMENT_NODE') && parent.shadowRoot) {
+            element = parent;
+            parent = parent.parentNode;
+        }
+        const index = parent && toArray(parent.childNodes).indexOf(element);
+        if (index !== null && index >= 0 && parent) {
+            const doc = editor.getDocument();
+            const range = doc.createRange();
+            range.setStart(parent, index);
+            range.collapse();
+            editor.setDOMSelection({
+                type: 'range',
+                range,
+                isReverted: false,
+            });
+        }
+    }
+
+    private keyDownHandler(editor: IEditor, event: KeyDownEvent) {
+        if (this.isEditing) {
+            if (event.rawEvent.key === 'Escape') {
+                this.removeImageWrapper();
+            } else {
+                const selection = editor.getDOMSelection();
+                const isImageSelection = selection?.type == 'image';
+                if (isImageSelection) {
+                    this.selectBeforeEditingImage(editor, selection.image);
+                }
+                this.applyFormatWithContentModel(
+                    editor,
+                    selection,
+                    this.isCropMode,
+                    isModifierKey(event.rawEvent) && isImageSelection //if it's a modifier key over a image, the image should select the image
+                );
+            }
+        }
+    }
+
+    private applyFormatWithContentModel(
+        editor: IEditor,
+        selection: DOMSelection | null,
```

Comment:
If the selection is always retrieved from editor, no need to pass it in, but get it from editor